iroh-net/src/magicsock.rs

```diff
@@ -1176,6 +1176,26 @@ impl Actor {
         out
     }
 
+    /// Split a number of transmits into individual packets.
+    ///
+    /// For each transmit, if it has a segment size, it will be split into
+    /// multiple packets according to that segment size. If it does not have a
+    /// segment size, the contents will be sent as a single packet.
+    fn split_packets(transmits: Vec<quinn_udp::Transmit>) -> Vec<Bytes> {
+        let mut res = Vec::with_capacity(transmits.len());
+        for transmit in transmits {
+            let contents = transmit.contents;
+            if let Some(segment_size) = transmit.segment_size {
+                for chunk in contents.chunks(segment_size) {
+                    res.push(contents.slice_ref(chunk));
```

Comment:
this `slice_ref` is probably a bit too lose for what we want.

But maybe rather than figure out what we're doing wrong we should start with copying Quinn's own implementation of this: https://github.com/quinn-rs/quinn/blob/f07a40d7f1da99253408fb1ab3db91eef3fe07e6/quinn-proto/src/tests/util.rs#L516-L537

Reply:
slice_ref just converts the `&[u8]` back to a `Bytes`. The question is if chunks does the right thing.

Reply:
This seems to be the exact same thing, except done in a loop. It is basically taking from the start, at most segment_size, until there is nothing left.

let end = segment_size.min(transmit.contents.len());

Reply:
oh, `slice_ref` uses pointer offsets.  that's probably fine then

Reply:
slice_ref is just some weird API to get around the fact that the APIs for slices is much richer than for Bytes, so you often have a &[u8] that you *know* is part of the original Bytes, and you don't want to use slice.to_vec().into() but do actual 0 copy.